cmd/seed-db/main.go

```diff
@@ -0,0 +1,215 @@
+									CountyName TEXT)`
+const citystateInsertQuery = `INSERT INTO city_state(CopyrightDetailCode,ZipCode,CityStateKey,ZipClassificationCode,CityStateName,CityStateNameAbbreviation,CityStateNameFacilityCode,CityStateMailingNameIndicator,PreferredLastLineCityStateKey,PreferredLastLineCityStateName,CityDeliveryIndicator,CarrierRouteRateSortation,UniqueZipNameIndicator,FinanceNumber,StateAbbreviation,CountyNumber,CountyName) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)`
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Fprintf(os.Stderr, "usage: %v <db-name>\n", filepath.Base(os.Args[0]))
+		os.Exit(1)
+	}
+
+	dbName := os.Args[1]
+
+	db, err := sql.Open("sqlite3", dbName)
+	if err != nil {
+		panic(err)
+	}
+	defer db.Close()
+
+	err = SeedZip4Data(db)
+	if err != nil {
+		panic(err)
+	}
+
+	err = SeedCityStateData(db)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func SeedZip4Data(db *sql.DB) error {
+	var zip4Data []zip4.Zip4Detail
+	count := 1
+	_, err := db.Exec(zip4CreateTableQuery)
+	if err != nil {
+		return err
+	}
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = zip4.ReadZip4FromZip4Tar("zip4natl.tar", mustGetenv("ZIP4_PWD"), func(detail zip4.Zip4Detail){
+		zip4Data = append(zip4Data, detail)
+		if count%BATCH_SIZE == 0 {
+			for i := 0; i < len(zip4Data); i++ {
+				params := getParameters(zip4Data[i])
+				_, err = tx.Exec(zip4InsertQuery, params...)
+				if err != nil {
+					panic(err)
+				}
+			}
+
+			zip4Data = []zip4.Zip4Detail{}
+		}
+		
+		count++
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(zip4Data) != 0 {
+		for i := 0; i < len(zip4Data); i++ {
+			params := getParameters(zip4Data[i])
+			_, err = tx.Exec(zip4InsertQuery, params...)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func SeedCityStateData(db *sql.DB) error {
+	var citystateData []citystate.CityStateDetail
+
+	_, err := db.Exec(citystateCreateTableQuery)
+	if err != nil {
+		return err
+	}
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = zip4.ReadCityStateFromZip4Tar("zip4natl.tar", mustGetenv("CITYSTATE_PWD"), func(detail citystate.CityStateDetail){
+		citystateData = append(citystateData, detail)
+		if len(citystateData)%BATCH_SIZE == 0 && len(citystateData) != 0 {
+			for i := 0; i < len(citystateData); i++ {
+				params := getParameters(citystateData[i])
+				_, err = tx.Exec(citystateInsertQuery, params...)
+				if err != nil {
+					panic(err)
+				}
+			}
+
+			citystateData = []citystate.CityStateDetail{}
+		}
+		
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(citystateData) != 0 {
+		for i := 0; i < len(citystateData); i++ {
+			params := getParameters(citystateData[i])
+			_, err = tx.Exec(citystateInsertQuery, params...)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getParameters(detail interface{}) []interface{} {
```

Comment:
Change occurrences of `interface{}` to `any`. The `any` keyword was added in go1.18.

Reply:
Done